test/FunctionalTests/Microsoft.OData.Core.Tests/Json/MockSyncOnlyJsonWriter.cs

```diff
@@ -0,0 +1,152 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MockSyncOnlyJsonWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Json;
+
+namespace Microsoft.OData.Tests.Json
+{
+    /// <summary>
+    /// A mock implementation of <see cref="IJsonWriter"/>
+    /// that does not implement the async interface <see cref="IJsonWriterAsync"/>
+    /// use for testing purposes.
+    /// 
+    /// Note: most methods are not actually implemented.
+    /// </summary>
+    internal class MockSyncOnlyJsonWriter : IJsonWriter
+    {
+        public void EndArrayScope()

```

Comment:
nit: You could keep the class shorter by using the shorter form, e.g., `public void EndArrayScope() => new NotImplementedException();`